src/extensions/extensionsHelper.ts

```diff
@@ -769,23 +923,48 @@ export async function uploadExtensionVersionFromLocalSource(args: {
   nonInteractive: boolean;
   force: boolean;
 }): Promise<ExtensionVersion | undefined> {
-  let extension;
+  const extensionRef = `${args.publisherId}/${args.extensionId}`;
+  let extension: Extension | undefined;
+  let latestVersion: ExtensionVersion | undefined;
   try {
-    extension = await getExtension(`${args.publisherId}/${args.extensionId}`);
+    extension = await getExtension(extensionRef);
+    latestVersion = await getExtensionVersion(`${extensionRef}@latest`);
   } catch (err: any) {
-    // Silently fail and continue the publish flow if extension not found.
+    // Silently fail and continue if extension is new or has no latest version set.
   }
+  displayExtensionHeader(extensionRef, extension, latestVersion?.extensionRoot);
 
-  const { extensionSpec, notes } = await validateExtensionSpec({
-    publisherId: args.publisherId,
-    extensionId: args.extensionId,
-    rootDirectory: args.rootDirectory,
-    latestVersion: extension?.latestVersion,
-    stage: args.stage,
-  });
+  const localStageOptions = ["rc", "alpha", "beta"];
+  if (args.stage && !localStageOptions.includes(args.stage)) {
+    throw new FirebaseError(
+      `--stage flag only supports the following values when used with --local: ${localStageOptions}`
```

Comment:
Same as above - super nit - this currently prints out with no spaces between the entries in stage options:
```suggestion
      `--stage flag only supports the following values when used with --local: ${localStageOptions.join(', ')}`
```